### ISSUE TITLE
Bare BIDS - can requirements for README be enhanced?

### ISSUE THREAD

**Issue by VisLab** [bids-standard/bids-specification#662]
Would it be possible to specify in more detail the requirements for the README file and to somehow validate that these requirements have been met?  The current spec reads:

> A README file SHOULD be found at the root of the sourcedata folder or the derivatives folder, or both. 
> This file should describe the nature of the raw data or the derived data. We RECOMMEND including the 
> PDF print-out with the actual sequence parameters generated by the scanner in the sourcedata folder.

As a reviewer I have encountered the issue of datasets being deposited into OpenNeuro with no useful descriptive information included. These datasets seem to have validated successfully and are uploaded, but they contain no description about what the experiment was about, what the task was, and what the goal was. The datasets contain no references to publications (even when there are some), essentially a blank README, and no auxiliary information and no JSON sidecars for describing the columns
of tsv files. The datasets are essentially useless for others.  This may not be as much of an issue when the experiments were mainly resting state  fMRI, but for modalities such as MEG and EEG, the information is essential. 

My questions:
1)   Could the requirements for the README be made more specific (maybe with specific issues to be addressed)?
2)   Could these requirements be validated in some way?

**Comment by effigies** [comment]
I suspect if it can be codified enough to be validated, it will make more sense as structured data rather than free text.

I do think guidelines are good, which could be useful for reviewers to point to when reviewing data, but I'm not sure that we can program cleverly enough to thwart a human determined to be unhelpful.

**Comment by sappelhoff** [comment]
> I suspect if it can be codified enough to be validated, it will make more sense as structured data rather than free text.

agreed.

However, one enhancement that could be validated that I could think of is to check the number of lines (or characters) in README, and warn if it seems like too little. The warning could then be along the lines:

> "Detected a very brief description of the data in README. It is highly RECOMMENDED to provide extensive information about the experiment in README, to facilitate re-use of the data."


Regarding this point:

> I do think guidelines are good, which could be useful for reviewers to point to when reviewing data,

agreed, and I think it would be in the scope of the BIDS specification to provide some guidelines of what constitutes a good README file. --> and if we were to find a too-short README during validation, the warning we send could include a link to those guidelines.

**Comment by Remi-Gau** [comment]
> agreed, and I think it would be in the scope of the BIDS specification to provide some guidelines of what constitutes a good README file.

Providing some templates on the side could also be nice for users as well.

**Comment by VisLab** [comment]
Here is a start on a template:

### Introduction

**Overview:** 
* A paragraph giving an overview of the experiment. This should include the goals or purpose and a discussion about how the experiment tries to achieve these goals.

* A brief overview of the tasks in the experiment.

* A brief discussion of condition variables (sometimes called contrasts or independent variables) 
that were varied across the experiment. 

* A brief discussion of the response variables (sometimes called the dependent variables) that were measured and or calculated to assess the effects of varying the condition variables. This might also include questionnaires administered to assess behavioral aspects of the experiment.

* A brief discussion of the control variables --- that is what aspects were explicitly controlled in this experiment. The control variables might include subject pool, environmental conditions, set up, or other things that were explicitly controlled.

More details on all of this should be included in the Methods section 


### Methods   

**Subjects:** A brief sentence about the subject pool in this experiment.
 
**Apparatus:**  A summary of the equipment and environment setup for the experiment. For example, 
was the experiment performed in a shielded room with the subject seated in a fixed position.

**Initial setup:** A summary of what setup was performed when a subject arrived.

**Task organization:** How the tasks were organized for a session. (This is particularly important because BIDS
studies usually separate task data into different files.) What was the order that tasks were performed? 
Was task order counter-balanced?  What other activities were interspersed because the tasks?  

**Task details:**  As much detail as possible about the task and the events that were recorded. 


**Additional data acquired:** 

**Experimental Location:** 

**Notes:**

**Comment by VisLab** [comment]
It might also be nice to have the validator put in a warning if no references and links are listed in the dataset_description.  Just a friendly message encouraging the data depositors to list publications that have used the data.  I'd actually like to see derivative publications beyond the "first" publications also listed.

**Comment by sappelhoff** [comment]
Cool, thanks for taking a stab at this @VisLab!

I suggest to proceed as follows:

- [ ] Add the template to https://github.com/bids-standard/bids-starter-kit/ in some sensible place (to be determined)
- [ ] make a release of the bids-starter kit, so that the link to the template becomes stable
- [ ] edit the bids-specification to point to the stable template link --> [edit this section here](https://bids-specification.readthedocs.io/en/latest/03-modality-agnostic-files.html#readme)
- [ ] make a PR to bids-validator so that a warning is raised if README is too small (exact definition of "small" to be determined)

regarding "warning" if no "references and links", please raise that issue in bids-validator @VisLab

**Comment by Remi-Gau** [comment]
LOVE this template already !!! 

Yes opening a PR on the starter kit makes a lot of sense. This way we can make suggestions and edits.

One of them that I thought of for Datalad curated dataset was to add a pointer to the Datalad template README: http://handbook.datalad.org/en/latest/basics/101-180-FAQ.html#how-can-i-help-others-get-started-with-a-shared-dataset

**Comment by Remi-Gau** [comment]
It would replace "template" README in the starter kit: 
https://github.com/bids-standard/bids-starter-kit/blob/master/templates/README

**Comment by VisLab** [comment]
As suggested, I have opened a pull request for the bids-starter-kit updating the README in the template.

**Comment by dorahermes** [comment]
This looks great.

@franklin-feingold, the templates in the bids-starter-kit will remain on github, correct? (asking because the wiki may get updated)

**Comment by Remi-Gau** [comment]
> This looks great.
> 
> @franklin-feingold, the templates in the bids-starter-kit will remain on github, correct? (asking because the wiki may get updated)

From my understanding is that even if things "move" in the BIDS starterkit, they will remain on github and only the link in the specs would have to change.

**Comment by franklin-feingold** [comment]
@dorahermes yep! As @Remi-Gau mentioned things may move around, but we are planning to stay on GitHub

**Comment by dorahermes** [comment]
Thanks!

**Comment by Remi-Gau** [comment]
> I suggest to proceed as follows:
> 
>     * [x]  Add the template to https://github.com/bids-standard/bids-starter-kit/ in some sensible place (to be determined)
> 
>     * [ ]  make a release of the bids-starter kit, so that the link to the template becomes stable
> 
>     * [ ]  edit the bids-specification to point to the stable template link --> [edit this section here](https://bids-specification.readthedocs.io/en/latest/03-modality-agnostic-files.html#readme)

Now that there is a README template, do we want to make a release of the starter-kit? @franklin-feingold Does it make sense to do that now before we start all the "restructuring" of the starter-kit?

Related issues:
- do we want to have the starter kit releases archived on zenodo too?

**Comment by Remi-Gau** [comment]
> Cool, thanks for taking a stab at this @VisLab!
> 
> I suggest to proceed as follows:
> 
>     * [x]  Add the template to https://github.com/bids-standard/bids-starter-kit/ in some sensible place (to be determined)
> 
>     * [ ]  make a release of the bids-starter kit, so that the link to the template becomes stable
> 
>     * [ ]  edit the bids-specification to point to the stable template link --> [edit this section here](https://bids-specification.readthedocs.io/en/latest/03-modality-agnostic-files.html#readme)
> 
>     * [ ]  make a PR to bids-validator so that a warning is raised if README is too small (exact definition of "small" to be determined)
> 
> 
> regarding "warning" if no "references and links", please raise that issue in bids-validator @VisLab

From our discussion in our last meeting I think that point 2 and 3 of that checklist are not the way we decided to do this, correct?

Though we might want to rephrase how the starter kit is mentioned in the landing page of the spec and maybe in the section for  about the README.

**Comment by sappelhoff** [comment]
> point 2 and 3 of that checklist are not the way we decided to do this, correct?

correct, we want to mention the starter-kit more vaguely (but still targeted and helpfully) instead of pointing to a specific stable (released) document.

In order to not over-complicate the starter-kit with versioning and technical issues --> that place should remain as accessible to new users as possible.

**Comment by Remi-Gau** [comment]
@sappelhoff This can be closed, no?
